### PR TITLE
Add cross validation loss calculation to training script

### PR DESCRIPTION
**Pull request by zhiying318**
This PR enables the calculation of **cross validation loss**. It is added to `train.py`. 

For now the cross validation loss is calculated **after each epoch**. If you'd like to make it calculate every 5 / 10 / .. epochs, I can add an `if` condition before the loggers.

p.s. If there's no validation split in the dataset, the cross validation part will be skipped.

### CODE REVIEW

**Review by ariG23498 — APPROVED** [review]
Thanks for doing this! LGTM
